mathics/builtin/calculus.py

```diff
@@ -962,6 +962,67 @@ def apply(self, expr, x, x0, evaluation, options={}):
             return from_sympy(result)
 
 
+class DiscreteLimit(Builtin):
+    """
+    <dl>
+    <dt>'DiscreteLimit[$f$, $k$->Infinity]'
+        <dd>gives the limit of the sequence $f$ as $k$ tends to infinity.
+    </dl>
+
+    >> DiscreteLimit[n/(n + 1), n -> Infinity]
+     = 1
+
+    >> DiscreteLimit[f[n], n -> Infinity]
+     = f[Infinity]
+    """
+
+    # TODO: Make this work
+    """
+    >> DiscreteLimit[(n/(n + 2)) E^(-m/(m + 1)), {m -> Infinity, n -> Infinity}]
+     = 1 / E
+    """
+
+    attributes = ('Listable',)
+
+    options = {
+        'PerformanceGoal': '"Quality"',
+    }
+
+    messages = {
+        'lperf': "Value of PerformanceGoal -> `1` should be \"Quality\" or \"Speed\".",
+    }
+
+    def apply(self, f, n, n0, evaluation, options={}):
+        'DiscreteLimit[f_, n_->n0_, OptionsPattern[DiscreteLimit]]'
+
+        f = f.to_sympy(convert_all_global_functions=True)
+        n = n.to_sympy()
+        n0 = n0.to_sympy()
+
+        if n0 != sympy.oo:
+            return
+
+        if f is None or n is None:
+            return
+
+        perf_goal = options['System`PerformanceGoal'].to_python()
+        if perf_goal == '"Quality"':
+            trials = 50
```

Comment:
These numbers are completely arbitrary. We should probably consider using the maximum number of iterations in here, or something along this lines.